diff --git a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleFunctionParser.java b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleFunctionParser.java
index 3aa0522..d66f9ff 100644
--- a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleFunctionParser.java
+++ b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleFunctionParser.java
@@ -3,7 +3,6 @@ package com.mapbox.rctmgl.components.styles;
 import android.util.Log;
 
 import com.facebook.react.bridge.Dynamic;
-import com.facebook.react.bridge.DynamicFromMap;
 import com.facebook.react.bridge.NoSuchKeyException;
 import com.facebook.react.bridge.ReadableArray;
 import com.facebook.react.bridge.ReadableMap;
diff --git a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/modules/RCTMGLModule.java b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/modules/RCTMGLModule.java
index c360777..5833f7f 100644
--- a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/modules/RCTMGLModule.java
+++ b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/modules/RCTMGLModule.java
@@ -39,7 +39,7 @@ import javax.annotation.Nullable;
 
 @ReactModule(name = RCTMGLModule.REACT_CLASS)
 public class RCTMGLModule extends ReactContextBaseJavaModule {
-    public static final String REACT_CLASS = "RCTMGLModule";
+    public static final String REACT_CLASS = "MGLModule";
 
     private static boolean customHeaderInterceptorAdded = false;
 
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/AbstractLayer.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/AbstractLayer.js
index 5d4aa9f..7b7a6f2 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/AbstractLayer.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/AbstractLayer.js
@@ -8,6 +8,11 @@ import {getStyleType} from '../utils/styleMap';
 import BridgeValue from '../utils/BridgeValue';
 
 class AbstractLayer extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.nativeLayer = React.createRef();
+  }
+
   get baseProps() {
     return {
       ...this.props,
@@ -62,12 +67,12 @@ class AbstractLayer extends React.PureComponent {
   }
 
   setNativeProps(props) {
-    if (this.refs.nativeLayer) {
+    if (this.nativeLayer.current) {
       let propsToPass = props;
       if (props.style) {
         propsToPass = {...props, reactStyle: this._getStyle(props.style)};
       }
-      this.refs.nativeLayer.setNativeProps(propsToPass);
+      this.nativeLayer.current.setNativeProps(propsToPass);
     }
   }
 }
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/AbstractSource.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/AbstractSource.js
index db8eac8..d38318f 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/AbstractSource.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/AbstractSource.js
@@ -4,10 +4,6 @@ class AbstractSource extends React.PureComponent {
   setNativeProps(props) {
     if (this._nativeRef) {
       this._nativeRef.setNativeProps(props);
-    } else {
-      if (this.refs.nativeSource) {
-        this.refs.nativeSource.setNativeProps(props);
-      }
     }
   }
 }
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/BackgroundLayer.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/BackgroundLayer.js
index e1cf1cf..4020a13 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/BackgroundLayer.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/BackgroundLayer.js
@@ -79,7 +79,7 @@ class BackgroundLayer extends AbstractLayer {
     return (
       <RCTMGLBackgroundLayer
         testID="rctmglBackgroundLayer"
-        ref="nativeLayer"
+        ref={this.nativeLayer}
         {...this.baseProps}
       />
     );
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/Camera.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/Camera.js
index 4dd55ea..d4087b9 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/Camera.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/Camera.js
@@ -196,6 +196,11 @@ class Camera extends React.Component {
     Linear: 'linearTo',
   };
 
+  constructor(props) {
+    super(props);
+    this.camera = React.createRef();
+  }
+
   UNSAFE_componentWillReceiveProps(nextProps) {
     this._handleCameraChange(this.props, nextProps);
   }
@@ -218,15 +223,15 @@ class Camera extends React.Component {
     }
 
     if (c.followUserLocation && !n.followUserLocation) {
-      this.refs.camera.setNativeProps({followUserLocation: false});
+      this.camera.current.setNativeProps({followUserLocation: false});
       return;
     }
     if (!c.followUserLocation && n.followUserLocation) {
-      this.refs.camera.setNativeProps({followUserLocation: true});
+      this.camera.current.setNativeProps({followUserLocation: true});
     }
 
     if (n.followUserLocation) {
-      this.refs.camera.setNativeProps({
+      this.camera.current.setNativeProps({
         followUserMode: n.followUserMode,
         followPitch: n.followPitch || n.pitch,
         followHeading: n.followHeading || n.heading,
@@ -236,17 +241,17 @@ class Camera extends React.Component {
     }
 
     if (n.maxBounds) {
-      this.refs.camera.setNativeProps({
+      this.camera.current.setNativeProps({
         maxBounds: this._getMaxBounds(),
       });
     }
     if (n.minZoomLevel) {
-      this.refs.camera.setNativeProps({
+      this.camera.current.setNativeProps({
         minZoomLevel: this.props.minZoomLevel,
       });
     }
     if (n.maxZoomLevel) {
-      this.refs.camera.setNativeProps({
+      this.camera.current.setNativeProps({
         maxZoomLevel: this.props.maxZoomLevel,
       });
     }
@@ -545,7 +550,7 @@ class Camera extends React.Component {
       cameraConfig = this._createStopConfig(config);
     }
 
-    this.refs.camera.setNativeProps({stop: cameraConfig});
+    this.camera.current.setNativeProps({stop: cameraConfig});
   }
 
   _createDefaultCamera() {
@@ -633,7 +638,7 @@ class Camera extends React.Component {
     return (
       <RCTMGLCamera
         testID="Camera"
-        ref="camera"
+        ref={this.camera}
         followUserLocation={this.props.followUserLocation}
         followUserMode={this.props.followUserMode}
         followPitch={this.props.followPitch}
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/CircleLayer.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/CircleLayer.js
index 1b414f3..0004e6a 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/CircleLayer.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/CircleLayer.js
@@ -83,7 +83,7 @@ class CircleLayer extends AbstractLayer {
     return (
       <RCTMGLCircleLayer
         testID="rctmglCircleLayer"
-        ref="nativeLayer"
+        ref={this.nativeLayer}
         {...this.baseProps}
       />
     );
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/FillExtrusionLayer.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/FillExtrusionLayer.js
index 8f581f5..4be1f43 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/FillExtrusionLayer.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/FillExtrusionLayer.js
@@ -83,7 +83,7 @@ class FillExtrusionLayer extends AbstractLayer {
       ...this.baseProps,
       sourceLayerID: this.props.sourceLayerID,
     };
-    return <RCTMGLFillExtrusionLayer ref="nativeLayer" {...props} />;
+    return <RCTMGLFillExtrusionLayer ref={this.nativeLayer} {...props} />;
   }
 }
 
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/FillLayer.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/FillLayer.js
index 8750b5e..1c45f34 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/FillLayer.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/FillLayer.js
@@ -83,7 +83,7 @@ class FillLayer extends AbstractLayer {
       ...this.baseProps,
       sourceLayerID: this.props.sourceLayerID,
     };
-    return <RCTMGLFillLayer ref="nativeLayer" {...props} />;
+    return <RCTMGLFillLayer ref={this.nativeLayer} {...props} />;
   }
 }
 
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/HeatmapLayer.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/HeatmapLayer.js
index 41e6e34..31d374d 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/HeatmapLayer.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/HeatmapLayer.js
@@ -84,7 +84,7 @@ class HeatmapLayer extends AbstractLayer {
       ...this.baseProps,
       sourceLayerID: this.props.sourceLayerID,
     };
-    return <RCTMGLHeatmapLayer ref="nativeLayer" {...props} />;
+    return <RCTMGLHeatmapLayer ref={this.nativeLayer} {...props} />;
   }
 }
 
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/ImageSource.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/ImageSource.js
index ae3eb94..0b79325 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/ImageSource.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/ImageSource.js
@@ -65,7 +65,7 @@ class ImageSource extends AbstractSource {
     };
 
     return (
-      <RCTMGLImageSource ref="nativeSource" {...props}>
+      <RCTMGLImageSource ref={ref => { this._nativeRef = ref; }} {...props}>
         {cloneReactChildrenWithProps(this.props.children, {
           sourceID: this.props.id,
         })}
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/Light.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/Light.js
index c7410c1..d127b7f 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/Light.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/Light.js
@@ -22,15 +22,15 @@ class Light extends AbstractLayer {
   };
 
   setNativeProps(props) {
-    if (this.refs.nativeLight) {
-      this.refs.nativeLight.setNativeProps(props);
+    if (this.nativeLayer.current) {
+      this.nativeLayer.current.setNativeProps(props);
     }
   }
 
   render() {
     return (
       <RCTMGLLight
-        ref="nativeLight"
+        ref={this.nativeLayer}
         testID="rctmglLight"
         {...this.props}
         style={undefined}
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/LineLayer.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/LineLayer.js
index d53e545..959a0d5 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/LineLayer.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/LineLayer.js
@@ -83,7 +83,7 @@ class LineLayer extends AbstractLayer {
       ...this.baseProps,
       sourceLayerID: this.props.sourceLayerID,
     };
-    return <RCTMGLLineLayer ref="nativeLayer" {...props} />;
+    return <RCTMGLLineLayer ref={this.nativeLayer} {...props} />;
   }
 }
 
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/RasterLayer.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/RasterLayer.js
index 471908e..f718b2b 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/RasterLayer.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/RasterLayer.js
@@ -80,7 +80,7 @@ class RasterLayer extends AbstractLayer {
       ...this.baseProps,
       sourceLayerID: this.props.sourceLayerID,
     };
-    return <RCTMGLRasterLayer ref="nativeLayer" {...props} />;
+    return <RCTMGLRasterLayer ref={this.nativeLayer} {...props} />;
   }
 }
 
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/RasterSource.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/RasterSource.js
index 9186456..d35c012 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/RasterSource.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/RasterSource.js
@@ -107,7 +107,7 @@ class RasterSource extends AbstractSource {
       attribution: this.props.attribution,
     };
     return (
-      <RCTMGLRasterSource ref="nativeSource" {...props}>
+      <RCTMGLRasterSource ref={ref => { this._nativeRef = ref; }} {...props}>
         {cloneReactChildrenWithProps(this.props.children, {
           sourceID: this.props.id,
         })}
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/SymbolLayer.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/SymbolLayer.js
index d1a036a..984c610 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/SymbolLayer.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/SymbolLayer.js
@@ -102,7 +102,7 @@ class SymbolLayer extends AbstractLayer {
     };
 
     return (
-      <RCTMGLSymbolLayer ref="nativeLayer" {...props}>
+      <RCTMGLSymbolLayer ref={this.nativeLayer} {...props}>
         {this.props.children}
       </RCTMGLSymbolLayer>
     );
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/VectorSource.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/VectorSource.js
index 291b385..94b5239 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/VectorSource.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/VectorSource.js
@@ -177,7 +177,7 @@ class VectorSource extends NativeBridgeComponent(AbstractSource) {
       onAndroidCallback: isAndroid() ? this._onAndroidCallback : undefined,
     };
     return (
-      <RCTMGLVectorSource ref="nativeSource" {...props}>
+      <RCTMGLVectorSource {...props}>
         {cloneReactChildrenWithProps(this.props.children, {
           sourceID: this.props.id,
         })}
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/components/annotations/Annotation.js b/node_modules/@react-native-mapbox-gl/maps/javascript/components/annotations/Annotation.js
index d8c6a47..4c85bac 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/components/annotations/Annotation.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/components/annotations/Annotation.js
@@ -39,6 +39,7 @@ class Annotation extends React.Component {
     };
 
     this.onPress = this.onPress.bind(this);
+    this.source = React.createRef();
   }
 
   componentDidUpdate(prevProps) {
@@ -104,7 +105,7 @@ class Annotation extends React.Component {
     return (
       <Animated.ShapeSource
         id={this.props.id}
-        ref="source"
+        ref={this.source}
         onPress={this.onPress}
         shape={this.state.shape}>
         {this.symbolStyle && (
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/index.js b/node_modules/@react-native-mapbox-gl/maps/javascript/index.js
index 3d2c65b..a650369 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/index.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/index.js
@@ -38,7 +38,7 @@ import AnimatedRouteCoordinatesArray from './utils/animated/AnimatedRouteCoordin
 import Style from './components/Style';
 import Logger from './utils/Logger';
 
-const MapboxGL = {...NativeModules.MGLModule};
+const MapboxGL = NativeModules.MGLModule || {};
 
 // static methods
 MapboxGL.requestAndroidLocationPermissions = async function () {
diff --git a/node_modules/@react-native-mapbox-gl/maps/javascript/utils/index.js b/node_modules/@react-native-mapbox-gl/maps/javascript/utils/index.js
index bc3c28f..6f566d9 100644
--- a/node_modules/@react-native-mapbox-gl/maps/javascript/utils/index.js
+++ b/node_modules/@react-native-mapbox-gl/maps/javascript/utils/index.js
@@ -1,12 +1,12 @@
 import React from 'react';
 import {
-  ViewPropTypes,
   View,
   NativeModules,
   findNodeHandle,
   Platform,
 } from 'react-native';
 import resolveAssetSource from 'react-native/Libraries/Image/resolveAssetSource';
+import {ViewPropTypes} from 'deprecated-react-native-prop-types'
 
 function getAndroidManagerInstance(module) {
   const haveViewManagerConfig =
