diff --git a/node_modules/react-native-fs/FS.common.js b/node_modules/react-native-fs/FS.common.js
index 62b7ba73..c9dcacfe 100755
--- a/node_modules/react-native-fs/FS.common.js
+++ b/node_modules/react-native-fs/FS.common.js
@@ -11,7 +11,9 @@ var RNFSManager = require('react-native').NativeModules.RNFSManager;
 
 var NativeEventEmitter = require('react-native').NativeEventEmitter;
 
-var RNFS_NativeEventEmitter = new NativeEventEmitter(RNFSManager);
+var RNFS_NativeEventEmitter = new NativeEventEmitter(
+  RNFSManager && RNFSManager.addListener && RNFSManager.removeListeners ? RNFSManager : null
+);
 
 var base64 = require('base-64');
 var utf8 = require('utf8');
diff --git a/node_modules/react-native-fs/android/src/main/java/com/rnfs/RNFSManager.java b/node_modules/react-native-fs/android/src/main/java/com/rnfs/RNFSManager.java
index 7aab307a..02bb2f0f 100755
--- a/node_modules/react-native-fs/android/src/main/java/com/rnfs/RNFSManager.java
+++ b/node_modules/react-native-fs/android/src/main/java/com/rnfs/RNFSManager.java
@@ -73,6 +73,10 @@ public class RNFSManager extends ReactContextBaseJavaModule {
   }
 
   private Uri getFileUri(String filepath, boolean isDirectoryAllowed) throws IORejectionException {
+    // Uri.parse expects encoded filepaths, but RNFS has been written to handle unencoded paths so encode first
+    if (filepath.startsWith("/")) {
+      filepath = Uri.encode(filepath, "/");
+    }
     Uri uri = Uri.parse(filepath);
     if (uri.getScheme() == null) {
       // No prefix, assuming that provided path is absolute path to file
