diff --git a/node_modules/react-native-hyperlink/dist/index.js b/node_modules/react-native-hyperlink/dist/index.js
index 37adff6..b29fb42 100644
--- a/node_modules/react-native-hyperlink/dist/index.js
+++ b/node_modules/react-native-hyperlink/dist/index.js
@@ -170,7 +170,7 @@ function (_Component) {
 Hyperlink.propTypes = {
   linkDefault: _propTypes["default"].bool,
   linkify: _propTypes["default"].object,
-  linkStyle: textPropTypes.style,
+  linkStyle: _propTypes["default"].any,
   linkText: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].func]),
   onPress: _propTypes["default"].func,
   onLongPress: _propTypes["default"].func,
