diff --git a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaProviderManager.kt b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaProviderManager.kt
index de6f685..328e746 100644
--- a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaProviderManager.kt
+++ b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaProviderManager.kt
@@ -5,16 +5,11 @@ import com.facebook.react.module.annotations.ReactModule
 import com.facebook.react.uimanager.ThemedReactContext
 import com.facebook.react.uimanager.UIManagerHelper
 import com.facebook.react.uimanager.ViewGroupManager
-import com.facebook.react.viewmanagers.RNCSafeAreaProviderManagerDelegate
 import com.facebook.react.viewmanagers.RNCSafeAreaProviderManagerInterface
 
 @ReactModule(name = SafeAreaProviderManager.REACT_CLASS)
 class SafeAreaProviderManager :
     ViewGroupManager<SafeAreaProvider>(), RNCSafeAreaProviderManagerInterface<SafeAreaProvider> {
-  private val mDelegate = RNCSafeAreaProviderManagerDelegate(this)
-
-  override fun getDelegate() = mDelegate
-
   override fun getName() = REACT_CLASS
 
   public override fun createViewInstance(context: ThemedReactContext) = SafeAreaProvider(context)
diff --git a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.kt b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.kt
index e45f210..ca44b8c 100644
--- a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.kt
+++ b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.kt
@@ -5,7 +5,6 @@ import com.facebook.react.module.annotations.ReactModule
 import com.facebook.react.uimanager.ReactStylesDiffMap
 import com.facebook.react.uimanager.StateWrapper
 import com.facebook.react.uimanager.ThemedReactContext
-import com.facebook.react.uimanager.ViewManagerDelegate
 import com.facebook.react.uimanager.annotations.ReactProp
 import com.facebook.react.viewmanagers.RNCSafeAreaViewManagerInterface
 import com.facebook.react.views.view.ReactViewGroup
@@ -15,10 +14,6 @@ import com.facebook.react.views.view.ReactViewManager
 class SafeAreaViewManager : ReactViewManager(), RNCSafeAreaViewManagerInterface<SafeAreaView> {
   override fun getName() = REACT_CLASS
 
-  // Make sure we're not using delegates for now since ReactViewGroupManager doesn't use one. If it
-  // does in the future we will need a way to compose delegates together.
-  override fun getDelegate(): ViewManagerDelegate<ReactViewGroup>? = null
-
   override fun createViewInstance(context: ThemedReactContext) = SafeAreaView(context)
 
   override fun createShadowNodeInstance() = SafeAreaViewShadowNode()
